pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -149,10 +148,16 @@ pub async fn download_layer_file<'a>(
         .with_context(|| format!("rename download layer file to {local_path}"))
         .map_err(DownloadError::Other)?;
 
```

Comment:
The comment also mentions that we should fsync the new name with `fsync(new)`. Perhaps the comment is wrong in this regard as the name is stored on the directory entry AFAIR?

Reply:
`fsync(new)` might be some ext3 with strange mode requirement, which we don't necessarily need to support. Could at least comment here the discrepency?

Reply:
> Perhaps the comment is wrong in this regard as the name is stored on the directory entry AFAIR?

The comment is wrong IMO, the second `fsync(new)` is not necessary if the first fsync is a true fsync and not an `fdatasync`.
I think Postgres's `file_utils.c` is also too conservative.

But, let's not have this discussion here.

I'm inclined to leave the comment unchanged and will revisit this at some later point / talk with postgres people why they think it's necessary.

Anyway, the rest of the pageserver code base also doesn't do the second `fsync()` whenever it needs to make created files durable, so, let's be consistent about it.